apps/charterafrica/src/payload/collections/Media.js

```diff
@@ -24,6 +24,12 @@ const Media = {
       ({ doc }) => ({
         ...doc,
         alt: doc.alt ?? null,
+        /*
+        Since we are setting ENV PAYLOAD_PUBLIC_APP_URL=http://localhost:3000 at build time, generating static pages(404,500) ends up with image urls with `http://localhost:3000/media/cfalogobw.svg`.
+        In this proposal, we are just returning pathnames without hostname since our next js apps and payload are served from the same instance.
+        */
+        src: new URL(doc.url).pathname,
+        url: new URL(doc.url).pathname,
       }),
```

Comment:
No need for two `new URL` calls; just create a proper function with `return`.
```suggestion
      ({ doc }) => {
        // Can't use absolute URLs in src because we set PAYLOAD_PUBLIC_APP_URL to localhost at build-time & will lead to 404 at runtime
        // Solution: use relative URLs
        const { pathname } = new URL(doc.url);
        return {
          ...doc,
          alt: doc.alt ?? null,
          src: pathname,
          url: pathname,
        }
      },
```

PS: This could even move to somewhere reusable since I assume it's something we're going to apply to all apps?

Reply:
> PS: This could even move to somewhere reusable since I assume it's something we're going to apply to all apps?
@kilemensi 
We can't somehow move this to packages, it tries to resolve other files with jsx syntax when you import in payload related files.
![image](https://github.com/user-attachments/assets/6fac75b5-37d9-433e-a587-7a661dd46f66)